File: Symulator Świata/grass.py
from plant import Plant


class Grass(Plant):

    def __init__(self, world, coord=None):
        self.name = 'g'
        self._strength = 0
        self._age = 0
        self._initiative = 0
        self.sprite = (0, 250, 0)
        self.world = world
        if coord is not None:
            world.board[coord] = self
        else:
            self.spawn()

    def createChild(self, coord):
        grass = Grass(self.world, coord)

File: PO PROJEKT 1/PO PROJEKT 1/PROJEKT PO PLIKI ŹRÓDŁOWE/Grass.cpp
#include "Grass.h"

Grass::Grass(World* world) {
	increaseStrength(0);
	setInitiative(0);
	sprite = 'g';
	this->world = world;
	spawn();
}

Grass::Grass(World* world, Coordinate coord) {
	increaseStrength(0);
	setInitiative(0);
	sprite = 'g';
	this->world = world;
	world->board.insert({ coord , this });
}

Organism* Grass::createChild(Coordinate coord) {
	return new Grass(world, coord);
}
